adal/src/main/java/com/microsoft/aad/adal/ADFSWebFingerValidator.java

```diff
@@ -0,0 +1,70 @@
+package com.microsoft.aad.adal;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Validates trusts between authorities and ADFS instances using DRS metadata and WebFinger.
+ */
+final class ADFSWebFingerValidator {
+
+    private ADFSWebFingerValidator() {
+        // utility class
+    }
+
+    /**
+     * Used for logging.
+     */
+    private static final String TAG = "AdfsWebFingerValidator";
```

Comment:
I prefer to do ADFSWebFingerValidator.class.getSimpleName, if later we need to rename it's easy to refactor.